app/src/main/java/org/hisp/dhis/android/dataentry/reports/ReportsAdapter.java

```diff
@@ -33,15 +34,19 @@
     @NonNull
     private final List<ReportViewModel> reportViewModels;
 
-    ReportsAdapter(@NonNull Context context) {
+    @NonNull
+    private final OnReportViewModelClickListener onClickListener;
+
+    ReportsAdapter(@NonNull Context context, @NonNull OnReportViewModelClickListener onClickListener) {
```

Comment:
At this point, I am not really sure what is the best way to do it. I propose to postpone this step up to the moment when I know how to handle this in idiomatic RxJava way.